src/Kunstmaan/AdminBundle/Helper/Security/Acl/AclHelper.php

```diff
@@ -166,40 +166,41 @@ private function getPermittedAclIdsSQLForUser(Query $query)
         }
 
         // Security context does not provide anonymous role automatically.
-        $uR = array('"IS_AUTHENTICATED_ANONYMOUSLY"');
+        $uR = array($stringQuoteChar . 'IS_AUTHENTICATED_ANONYMOUSLY' . $stringQuoteChar);
 
         foreach ($userRoles as $role) {
             // The reason we ignore this is because by default FOSUserBundle adds ROLE_USER for every user
             if (is_string($role)) {
                 if ($role !== 'ROLE_USER') {
-                    $uR[] = '"' . $role . '"';
+                    $uR[] = $stringQuoteChar . $role . $stringQuoteChar;
                 }
             } else {
                 // Symfony 3.4 compatibility
                 if ($role->getRole() !== 'ROLE_USER') {
-                    $uR[] = '"' . $role->getRole() . '"';
+                    $uR[] = $stringQuoteChar . $role->getRole() . $stringQuoteChar;
                 }
             }
         }
         $uR = array_unique($uR);
         $inString = implode(' OR s.identifier = ', $uR);
 
         if (\is_object($user)) {
-            $inString .= ' OR s.identifier = "' . str_replace(
-                '\\',
-                '\\\\',
-                \get_class($user)
-            ) . '-' . $user->getUserName() . '"';
+            $inString .= ' OR s.identifier = ' . $stringQuoteChar . get_class($user) . '-' . $user->getUserName() . $stringQuoteChar;
+        }
+
+        $objectIdentifierColumn = 'o.object_identifier';
+        if ($aclConnection->getDatabasePlatform()->getName() === 'postgresql') {
+            $objectIdentifierColumn = 'o.object_identifier::BIGINT';
         }
 
         $selectQuery = <<<SELECTQUERY
-SELECT DISTINCT o.object_identifier as id FROM {$databasePrefix}acl_object_identities as o
-INNER JOIN {$databasePrefix}acl_classes c ON c.id = o.class_id
-LEFT JOIN {$databasePrefix}acl_entries e ON (
+SELECT DISTINCT {$objectIdentifierColumn} as id FROM acl_object_identities as o
+INNER JOIN acl_classes c ON c.id = o.class_id
+LEFT JOIN acl_entries e ON (
     e.class_id = o.class_id AND (e.object_identity_id = o.id
     OR {$aclConnection->getDatabasePlatform()->getIsNullExpression('e.object_identity_id')})
 )
-LEFT JOIN {$databasePrefix}acl_security_identities s ON (
+LEFT JOIN acl_security_identities s ON (
 s.id = e.security_identity_id
 )
 WHERE c.class_type = {$rootEntity}
```

Comment:
Do we need quotes here? https://github.com/Kunstmaan/KunstmaanBundlesCMS/pull/1778/files#diff-2ae42119f573a6a61e26009641aa186d97e0f77b951dc02385d67d9daa075854R193